pylint/checkers/refactoring/implicit_booleaness_checker.py

```diff
@@ -324,9 +327,91 @@ def _implicit_booleaness_message_args(
         elif isinstance(target_node, (nodes.Attribute, nodes.Name)):
             instance_name = target_node.as_string()
         original_comparison = f"{instance_name} {operator} {collection_literal}"
-        suggestion = f"{instance_name}" if operator == "!=" else f"not {instance_name}"
+        suggestion = self._get_suggestion(node, instance_name, operator, {"!="})
         return original_comparison, suggestion, description
 
+    def _get_suggestion(
+        self,
+        node: nodes.Compare,
+        name: str,
+        operator: str,
+        negation_redundant_ops: set[str],
+    ) -> str:
+        if operator in negation_redundant_ops:
+            return f"{name}" if self._in_boolean_context(node) else f"bool({name})"
+        return f"not {name}"
+
+    def _in_boolean_context(self, node: nodes.Compare) -> bool:
+        """
+        Returns True if the comparison is used in a boolean context; False otherwise.
+
+        A comparison is considered to be in a boolean context when it appears in constructs
+        that evaluate its truthiness directly, such as:
+        - control flow statements (`if`, `while`, `assert`)
+        - ternary expressions (`x if cond else y`)
+        - logical negation (`not`)
+        - comprehension filters (e.g. `[x for x in items if cond]`)
+        - generator expressions passed to `all()` or `any()`
+        - lambdas expressions passed to `filter()`
+        - `bool()` cast
+        - boolean operations `and`, `or` nested within any of the above
+
+        In contrast, a comparison is not in a boolean context when its result is used as a value,
+        such as when it is assigned to a variable, returned from a function, passed as an argument,
+        or used in an expression that does not depend on its truthiness.
+        """
```

Comment:
This function could probably be used elsewhere (or exists elsewhere ?), i.e. become a helper now or later.

Reply:
good point, i think this function is generic enough to be placed in utils for reusability